controller/pods/controller.go

```diff
@@ -451,10 +451,16 @@ func (p *PodInfo) PodToConsulService(containerStatus v1.ContainerStatus, cfg *co
 	service.Address = p.IP
 
 	if p.isProbeLivenessEnabled() {
-		service.Checks = append(service.Checks, p.probeToConsulCheck(p.getContainerLivenessProbe(containerStatus.Name), "Liveness Probe"))
+		check := p.probeToConsulCheck(p.getContainerLivenessProbe(containerStatus.Name), "Liveness Probe")
```

Comment:
@LoveBeforT I am trying to figure out a bug here as well. Maybe you are fixing the same thing.

It seems that when I define a port alias:

```yaml
          ports:
            - name: http
              containerPort: 8080
              protocol: TCP
```

I can use them in a `livenessProbe` (or `readinessProbe`):

```yaml
          livenessProbe:
            httpGet:
              path: /alive
              port: http
            initialDelaySeconds: 30
```

However the code in `kube-consul-register` somewhere/somehow evaluates the bit to `0` (Instead of looking it up in `ports`?) and then the service check fails in consul. Are you maybe after something similar?

Reply:
I debugged this a bit further (by adding more logs):

```
I0807 12:05:26.831311       1 consul.go:99] Registering service cadvisor-rj45s with ID: cadvisor-rj45s-cadvisor
I0807 12:05:26.831322       1 consul.go:101] Address: 10.244.147.42:8080
I0807 12:05:26.831325       1 consul.go:103] Checks (total): 1
I0807 12:05:26.831327       1 consul.go:107] Check ID: 
I0807 12:05:26.831330       1 consul.go:108] Check (HTTP): 
I0807 12:05:26.831333       1 consul.go:109] Check (SHELL): 
I0807 12:05:26.831336       1 consul.go:110] Check (TCP): 
I0807 12:05:26.831339       1 consul.go:111] Check (all fields) &api.AgentServiceCheck{CheckID:"", Name:"", Args:[]string(nil), DockerContainerID:"", Shell:"", Interval:"", Timeout:"", TTL:"", HTTP:"", Header:map[string][]string(nil), Method:"", TCP:"", Status:"", Notes:"", TLSSkipVerify:false, GRPC:"", GRPCUseTLS:false, AliasNode:"", AliasService:"", DeregisterCriticalServiceAfter:""}
```

Reply:
(Sorry for the noise)

But I think I just figured it out, there are essentially two bugs in the code:

- the `nil` on a probe causes the error
- there is a wrong assumption around "pods having probes", but really containers within a pod do

Can't blame anyone for this, I just (re-)learned the whole thing where a POD maybe n different containers, etc.. I keep forgetting this as well.